recipes/sbp/all/conanfile.py

```diff
@@ -0,0 +1,70 @@
+import os
+
+from conans import ConanFile, CMake, tools
+from conans.errors import ConanInvalidConfiguration
+
+
+class SbpConan(ConanFile):
+    name = "sbp"
+    license = "MIT"
+    homepage = "https://github.com/swift-nav/libsbp"
+    url = "https://github.com/conan-io/conan-center-index"
+    description = "Swift Binary Protocol client library"
+    topics = ("gnss",)
+    settings = "os", "compiler", "build_type", "arch"
+    options = {"shared": [True, False], "fPIC": [True, False]}
+    default_options = {"shared": False, "fPIC": True}
+    generators = "cmake"
+    exports_sources = "CMakeLists.txt", "c"
+
+    _cmake = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+
+        if self.settings.os == "Windows" and self.options.shared:
+            raise ConanInvalidConfiguration("Windows shared builds are not supported right now, see issue https://github.com/swift-nav/libsbp/issues/1062")
+
+    def source(self):
+        data = self.conan_data["sources"][self.version]
+
+        source = {"url": data["url"]["source"], "sha256": data["sha256"]["source"]}
+        tools.get(**source, strip_root=True, destination=self._source_subfolder)
+
+        cmake = {"url": data["url"]["cmake"], "sha256": data["sha256"]["cmake"]}
+        tools.get(**cmake, strip_root=True, destination=os.path.join(self._source_subfolder, "c", "cmake", "common"))
+
+    def _configure_cmake(self):
+        if self._cmake:
+            return self._cmake
+
+        self._cmake = CMake(self)
```

Comment:
Because examples are not installed, we try to not build them.
It looks like it is not possible to disable the examples.
See https://github.com/swift-nav/libsbp/blob/f4f6c8547ff0aa9f7625c7d49432dff0aefaa42f/c/CMakeLists.txt#L36
I propose adding a patch adding such an option or force disabling.
e.g.
replacing
```python
add_subdirectory(examples)
```

with
```python
#add_subdirectory(examples)
```
OR
```python
option(libbsp_BUILD_EXAMPLES "Build libbsp examples" ON)
if(libbsp_BUILD_EXAMPLES)
add_subdirectory(examples)
endif()
```

I vouch for the latter, as this will not change the default behavior and can be upstreamed.

Reply:
The examples are not built by default.  See https://github.com/swift-nav/libsbp/blob/master/c/examples/CMakeLists.txt

Reply:
Oh, great then!